test/e2e/framework/service/jig.go

```diff
@@ -82,6 +96,66 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocate tries to allocate the passed node port and
+// return true if succeeds or false if can not be allocated.
+// Should not be used directly as it does not have lock protection and hence avoids deadlock.
+// Should only be called from a function that has acquired lock on nodeports
+func (s *staticPortRange) allocate(port int) bool {
+	// port is out of range, it can not be allocated
```

Comment:
```suggestion
         s.mu.Lock()
	defer s.mu.Unlock()
	// port is out of range, it can not be allocated
```

and remove the lock in allocateNext

Reply:
hmm I understand now